### PR TITLE
CASMINST-5441 Use local helm config to inherit authentication to csm-helm-charts

### PR DESCRIPTION
**Pull request by mtupitsyn**
## Summary and Scope

CSM Helm repo at https://artifactory.algol60.net/artifactory/csm-helm-charts/ will require authentication soon. To avoid HTTP 401 errors while running `helm dependency update` command, I'd suggest to inherit default Helm config directory from the host which runs helm docker container. Thus, if authentication is already made at the host, it will be inherited by container.

This will cover builds in Jenkins (as Jenkins workers already pre-authenticate to private helm repository), and allow to inherit authentication for local builds. I.e. one just need to run

    helm repo add csm-algol60 https://artifactory.algol60.net/artifactory/csm-helm-charts/ --username <your_username> --password-stdin

to establish authentication locally, and it will be inherited by helm running in container.

## Issues and Related PRs

* Needed for [CASMINST-5441](https://jira-pro.its.hpecorp.net:8443/browse/CASMINST-5441)

### Tested on:

  * Local development environment
  * Jenkins development build

### Test description:

* Alternate private Helm repo was created at https://artifactory.algol60.net/artifactory/csm-helm-charts-private/
* Dependency chart `cray-service-7.0.2.tgz` placed in private repo
* Dependency link in `Chart.yaml` temporarily changed to point to private repo
* Build is successful:
https://jenkins.algol60.net/job/Cray-HPE/job/cray-sts/job/feature%252Fhelm-auth/3/console

      14:33:23  docker run --rm \
      14:33:23      --user 10000:10000 \
      14:33:23      --mount type=bind,src="/home/jenkins/workspace/y-HPE_cray-sts_feature_helm-auth",dst=/src \
      14:33:23      --mount type=bind,src=/home/jenkins/.config/helm,dst=/tmp/.helm/config \
      14:33:23      -w /src \
      14:33:23      -e HELM_CACHE_HOME=/src/.helm/cache \
      14:33:23      -e HELM_CONFIG_HOME=/tmp/.helm/config \
      14:33:23      -e HELM_DATA_HOME=/src/.helm/data \
      14:33:23      artifactory.algol60.net/csm-docker-private/docker.io/alpine/helm:3.7.1 \
      14:33:23      dep up kubernetes/cray-sts

## Risks and Mitigations

Low - build time change only

### CODE REVIEW

File: Jenkinsfile.github [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 pipeline {
     agent {
-        label "metal-gcp-builder"
+        label "metal-gcp-builder-dev"
```
Comment (mtupitsyn) [inline Jenkinsfile.github:5]: Temporary change, needed for testing

File: Makefile [hunk 2]
```diff
@@ -2,15 +2,21 @@ NAME ?= cray-sts
 VERSION ?= $(shell cat .version)
 
 CHART_VERSION ?= $(VERSION)
-IMAGE ?= artifactory.algol60.net/csm-docker/stable/${NAME}
+IMAGE ?= artifactory.algol60.net/csm-docker-private/stable/${NAME}
```
Comment (mtupitsyn) [inline Makefile:5]: Temporary change, needed for testing

File: Makefile [hunk 3]
```diff
@@ -2,15 +2,21 @@ NAME ?= cray-sts
...
-CHART_METADATA_IMAGE ?= artifactory.algol60.net/csm-docker/stable/chart-metadata
-YQ_IMAGE ?= artifactory.algol60.net/docker.io/mikefarah/yq:4
-HELM_IMAGE ?= artifactory.algol60.net/docker.io/alpine/helm:3.7.1
-HELM_UNITTEST_IMAGE ?= artifactory.algol60.net/docker.io/quintush/helm-unittest
-HELM_DOCS_IMAGE ?= artifactory.algol60.net/docker.io/jnorwood/helm-docs:v1.5.0
+CHART_METADATA_IMAGE ?= artifactory.algol60.net/csm-docker-private/stable/chart-metadata
+YQ_IMAGE ?= artifactory.algol60.net/csm-docker-private/docker.io/mikefarah/yq:4
+HELM_IMAGE ?= artifactory.algol60.net/csm-docker-private/docker.io/alpine/helm:3.7.1
+HELM_UNITTEST_IMAGE ?= artifactory.algol60.net/csm-docker-private/docker.io/quintush/helm-unittest
+HELM_DOCS_IMAGE ?= artifactory.algol60.net/csm-docker-private/docker.io/jnorwood/helm-docs:v1.5.0
```
Comment (mtupitsyn) [inline Makefile:13]: All changes above are to verify that images can be successfully pulled from private registry during build. They are all temporary.

File: kubernetes/cray-sts/Chart.yaml [hunk 4]
```diff
@@ -8,7 +8,7 @@ home: https://github.com/Cray-HPE/cray-sts
 dependencies:
   - name: cray-service
     version: ~7.0.0
-    repository: https://artifactory.algol60.net/artifactory/csm-helm-charts/
+    repository: https://artifactory.algol60.net/artifactory/csm-helm-charts-private/
```
Comment (mtupitsyn) [inline kubernetes/cray-sts/Chart.yaml:11]: Temporary change, needed for testing

File: Jenkinsfile.github (hunk 1 above)
Comment (bklei) [reply Jenkinsfile.github]: removed

File: Makefile (hunk 3 above)
Comment (bklei) [reply Makefile]: removed

File: kubernetes/cray-sts/Chart.yaml (hunk 4 above)
Comment (bklei) [reply kubernetes/cray-sts/Chart.yaml]: removed

File: Makefile (hunk 2 above)
Comment (bklei) [reply Makefile]: removed

**Pull request merged**